src/eth/primitives/block_header.rs

```diff
@@ -48,10 +63,21 @@ impl BlockHeader {
             number,
             hash: number.hash(),
             transactions_root: HASH_EMPTY_TRANSACTIONS_ROOT,
-            gas: Gas::ZERO,
+            gas_used: Gas::ZERO,
+            gas_limit: Gas::MAX,
             bloom: LogsBloom::default(),
             timestamp,
             parent_hash: number.prev().map(|n| n.hash()).unwrap_or(Hash::zero()),
+            author: Address::default(),
+            extra_data: Bytes::default(),
+            miner: Address::default(),
+            difficulty: Difficulty::default(),
+            receipts_root: Hash::zero(),
+            uncle_hash: Hash::zero(),
```

Comment:
Empty hashes usually have special values like the ones defined here instead of being zero.

It is important for this to be correct because libraries like the Geth client in Go validates this information.

https://github.com/cloudwalk/stratus/blob/f3216247ee402489f53005951325fb865337635f/src/eth/primitives/block_header.rs#L27-L31

Reply:
Roger